go/adbc/driver/flightsql/flightsql_adbc.go

```diff
@@ -103,6 +103,12 @@ func (d Driver) NewDatabase(opts map[string]string) (adbc.Database, error) {
 		return nil, adbc.Error{Msg: err.Error(), Code: adbc.StatusInvalidArgument}
 	}
 
+	if db.uri.Scheme == "grpc+tls" {
+		db.creds = credentials.NewTLS(&tls.Config{})
+	} else {
+		db.creds = insecure.NewCredentials()
+	}
```

Comment:
should this get shifted into `SetOptions` since we do the rest of the handling for creds there?

Reply:
Done.